evm/src/logic.rs

```diff
@@ -0,0 +1,304 @@
+use std::marker::PhantomData;
+
+use plonky2::field::extension_field::{Extendable, FieldExtension};
+use plonky2::field::packed_field::PackedField;
+use plonky2::hash::hash_types::RichField;
+
+use crate::constraint_consumer::{ConstraintConsumer, RecursiveConstraintConsumer};
+use crate::stark::Stark;
+use crate::vars::{StarkEvaluationTargets, StarkEvaluationVars};
+
+// Total number of bits per input/output.
+const VAL_BITS: usize = 256;
+// Number of bits stored per field element. Ensure that this fits; it is not checked.
+const PACKED_LIMB_BITS: usize = 16;
+// Number of field elements needed to store each input/output at the specified packing.
+const PACKED_LEN: usize = (VAL_BITS + PACKED_LIMB_BITS - 1) / PACKED_LIMB_BITS;
+// Number of bits in the last limb, which may be smaller than `PACKED_LIMB_BITS`.
+const LAST_LIMB_BITS: usize = VAL_BITS - PACKED_LIMB_BITS * (PACKED_LEN - 1);
+
+pub(crate) mod columns {
+    use std::ops::Range;
+
+    use super::{PACKED_LEN, VAL_BITS};
+
+    pub const IS_AND: usize = 0;
+    pub const IS_OR: usize = IS_AND + 1;
+    pub const IS_XOR: usize = IS_OR + 1;
+    pub const AND_COEFF: usize = IS_XOR + 1;
```

Comment:
It seems like this isn't needed (since it's degree 1) and you didn't end up using it?

Reply:
Good catch! That’s a remnant of an old approach that I later dumped :)